From: Dmitry V. Levin
Date: Sun, 22 Nov 2009 13:48:15 +0000
Subject: [PATCH] tests: do not fail on read-only root file system

* tests/touch/not-owner: Handle the case when the root file system is
mounted read-only.
Reported by Solar Designer.
---
 tests/touch/not-owner |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/tests/touch/not-owner b/tests/touch/not-owner
index 3dd8a80..9cfa026 100755
--- a/tests/touch/not-owner
+++ b/tests/touch/not-owner
@@ -42,7 +42,10 @@ touch / > out 2>&1 && fail=1
 
 # On SunOS4, EPERM is `Not owner'.
 # On some *BSD systems it's `Operation not permitted'.
-for msg in 'Not owner' 'Operation not permitted' 'Permission denied'; do
+# On a system where root file system is mounted read-only
+# it's `Read-only file system'.
+for msg in 'Not owner' 'Operation not permitted' 'Permission denied' \
+           'Read-only file system'; do
   cat > exp <<EOF
 touch: setting times of \`/': $msg
 EOF
